web/html/src/manager/state/recurring-states-edit.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable */
+'use strict';
+
+const React = require("react");
+const ReactDOM = require("react-dom");
+const {RecurringEventPicker} = require("components/recurring-event-picker");
+const {DisplayHighstate} = require("./display-highstate");
+const Button = require("components/buttons").Button;
+const AsyncButton = require("components/buttons").AsyncButton;
+const {Toggler} = require("components/toggler");
+const { InnerPanel } = require("components/panels/InnerPanel");
+
+class RecurringStatesEdit extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            minions: minions,
+            active: true
+        };
+
+        if(this.isEdit()) {
+            this.setSchedule(this.props.schedule);
+        } else {
+            this.getTargetType();
+        }
+    }
+
+    setSchedule = (schedule) => {
+        Object.assign(this.state, schedule);
+    };
+
+    getTargetType = () => {
+        const search = window.location.search;
+        if (search.match("\\?sid")) {
```

Comment:
I'd like to have a field or a property populated from the backend instead, relying on a slice of the URL is not really robust from my POV, the URL could also change at any time.